pip/baseparser.py

```diff
@@ -145,23 +145,39 @@ def check_default(self, option, key, val):
             print("An error occurred during configuration: %s" % exc)
             sys.exit(3)
 
+    def _get_ordered_configuration_items(self):
+        # Configuration gives keys in an unordered manner. Order them.
+        override_order = ["global", self.name, ":env:"]
+
+        # Pool the options into different groups
+        section_items = {name: [] for name in override_order}
+        for key, val in self.config.items():
```

Comment:
Maybe use `for section_key, val in self.config.items():` here and `section, key = section_key.split(".", 1)` a line below